### ISSUE TITLE
Code escaping/highlighting bug

### ISSUE THREAD

**Issue by humphd** [Seneca-CDOT/telescope#1801]
Noticed reading [this post](https://dev.to/chrispinkney/il-meglio-e-l-inimico-del-bene-25f2) from @chrispinkney:

<img width="1465" alt="Screen Shot 2021-02-20 at 9 51 32 AM" src="https://user-images.githubusercontent.com/427398/108599789-46190980-7361-11eb-8a09-0c9e08ffdea8.png">

**Comment by HyperTHD** [comment]
Reading the same post today on dev, and I do not see the `Enter fullscreen mode Exit fullscreen mode` text on the page at any point in the post. May have been solved accidently at some point

**Comment by chrispinkney** [comment]
> 
> 
> Reading the same post today on dev, and I do not see the `Enter fullscreen mode Exit fullscreen mode` text on the page at any point in the post. May have been solved accidently at some point

Was the code visible though? I think the issue was created because of the code not showing on the Telescope version. Talking about this: 
```
<>
    <MDXPageBase title='About Us'>
        <About />
    </MDXPageBase>
</>
```

**Comment by humphd** [comment]
It's the React code snippet this bug is about.

**Comment by Metropass** [comment]
Okay, so after hours of debugging, here's what I found:

Our react code gets sanitized by our sanitizer in `sanitize.js`

![Screenshot_399](https://user-images.githubusercontent.com/58116522/113949938-238e6300-97de-11eb-9ba5-936d9ad4631e.png)


Going deep into debugging, we notice that the code is then passed into TypeScript's `Parser.write()` function, with `chunk` being our React code. `chunk` is still in it's original state, and nothing has been changed.

![Screenshot_405](https://user-images.githubusercontent.com/58116522/113950166-b3cca800-97de-11eb-9c69-46eed80b2911.png)

When we go into the write function, it's called through `Tokenizer`'s `write()` 
![Screenshot_406](https://user-images.githubusercontent.com/58116522/113950612-c4315280-97df-11eb-845a-f2ecb385af42.png)

After which, it parses the string letter by letter, and checks what the character is classified under.
![Screenshot_407](https://user-images.githubusercontent.com/58116522/113950718-fd69c280-97df-11eb-8c4b-873df3c63f6d.png)

Now, the function actually does detect that the string `MDXPageBase` is a type of tag in this line here.
![Screenshot_408](https://user-images.githubusercontent.com/58116522/113950822-4752a880-97e0-11eb-887d-e0e7ea917d40.png)

and it also detects the attribute name `title` as well. 
![Screenshot_409](https://user-images.githubusercontent.com/58116522/113950925-82ed7280-97e0-11eb-855b-71ee73bb52bc.png)

And it's value.
![Screenshot_410](https://user-images.githubusercontent.com/58116522/113951312-52f29f00-97e1-11eb-9836-5471808ecdff.png)

When the Tokenizer is finished parsing the data, `chunk` still stays the same HOWEVER, once we exit out of the function and we write to our String, (we're back on SanitizeHTML's code right now) on this line, we lose our data. 

![Screenshot_412](https://user-images.githubusercontent.com/58116522/113951633-ec21b580-97e1-11eb-8acb-edb700b41568.png)

This is a bug that is very, very complex. I'll look further into this, but it's good to document this here.

**Comment by chrispinkney** [comment]
Jeez, really getting into the nitty and gritty of the library eh? Is this not something that a simple config edit or setting change will fix for us, as opposed to a PR to the `sanitize-html` library?

**Comment by chrispinkney** [comment]
I made a quick search and [found this](https://www.npmjs.com/package/sanitize-html-react) library (which will replace sanitize-html): 

>This is slightly modified version of sanitize-html made for React. It no longer escapes the < > " & characters because React already escapes for you as described here.

Perhaps the problem here is because the sanitizer is removing the react fragments. Maybe this could be a solution?

**Comment by Metropass** [comment]
That's a good find! I'll check it out today, and post the results!

**Comment by Metropass** [comment]
So, that package is quite unsafe:
https://mobeigi.com/blog/security/xss/sanitize-html-react-vulnerability/

I'll test out another package called `DOMPurify`
https://www.npmjs.com/package/dompurify

**Comment by humphd** [comment]
Fixed by #2337
